src/gui/MainWindow.cpp

```diff
@@ -2190,55 +2190,30 @@ void MainWindow::initActionCollection()
                     m_ui->actionBugReport,
                     m_ui->actionAbout});
 
-    // Add actions whose shortcuts were set in the .ui file
+    // Register as default any shortcuts that were set in the .ui file
     for (const auto action : ac->actions()) {
         if (!action->shortcut().isEmpty()) {
             ac->setDefaultShortcut(action, action->shortcut());
         }
     }
 
-    // Actions with standard shortcuts
-    ac->setDefaultShortcut(m_ui->actionDatabaseOpen, QKeySequence::Open, Qt::CTRL + Qt::Key_O);
-    ac->setDefaultShortcut(m_ui->actionDatabaseSave, QKeySequence::Save, Qt::CTRL + Qt::Key_S);
-    ac->setDefaultShortcut(m_ui->actionDatabaseSaveAs, QKeySequence::SaveAs, Qt::CTRL + Qt::SHIFT + Qt::Key_S);
-    ac->setDefaultShortcut(m_ui->actionDatabaseClose, QKeySequence::Close, Qt::CTRL + Qt::Key_W);
-    ac->setDefaultShortcut(m_ui->actionSettings, QKeySequence::Preferences, Qt::CTRL + Qt::Key_Comma);
-    ac->setDefaultShortcut(m_ui->actionQuit, QKeySequence::Quit, Qt::CTRL + Qt::Key_Q);
-    ac->setDefaultShortcut(m_ui->actionEntryNew, QKeySequence::New, Qt::CTRL + Qt::Key_N);
+    // Actions with standard shortcuts (if no standard shortcut exists, leave the existing
+    // shortcuts from the .ui file in place)
+    ac->setDefaultShortcut(m_ui->actionDatabaseOpen, QKeySequence::Open);
+    ac->setDefaultShortcut(m_ui->actionDatabaseSave, QKeySequence::Save);
+    ac->setDefaultShortcut(m_ui->actionDatabaseSaveAs, QKeySequence::SaveAs);
+    ac->setDefaultShortcut(m_ui->actionDatabaseClose, QKeySequence::Close);
+    ac->setDefaultShortcut(m_ui->actionSettings, QKeySequence::Preferences);
+    ac->setDefaultShortcut(m_ui->actionQuit, QKeySequence::Quit);
+    ac->setDefaultShortcut(m_ui->actionEntryNew, QKeySequence::New);
 
     // Prevent conflicts with global Mac shortcuts (force Control on all platforms)
+    // Note: Qt::META means Ctrl on Mac.
 #ifdef Q_OS_MAC
-    auto modifier = Qt::META;
-#else
-    auto modifier = Qt::CTRL;
+    ac->setDefaultShortcut(m_ui->actionEntryAddToAgent, Qt::META + Qt::Key_H);
+    ac->setDefaultShortcut(m_ui->actionEntryRemoveFromAgent, Qt::META + Qt::SHIFT + Qt::Key_H);
 #endif
 
-    // All other actions with default shortcuts
-    ac->setDefaultShortcut(m_ui->actionDatabaseNew, Qt::CTRL + Qt::SHIFT + Qt::Key_N);
-    ac->setDefaultShortcut(m_ui->actionDatabaseSettings, Qt::CTRL + Qt::SHIFT + Qt::Key_Comma);
-    ac->setDefaultShortcut(m_ui->actionReports, Qt::CTRL + Qt::SHIFT + Qt::Key_R);
-    ac->setDefaultShortcut(m_ui->actionLockDatabase, Qt::CTRL + Qt::Key_L);
-    ac->setDefaultShortcut(m_ui->actionLockAllDatabases, Qt::CTRL + Qt::SHIFT + Qt::Key_L);
-    ac->setDefaultShortcut(m_ui->actionEntryEdit, Qt::CTRL + Qt::Key_E);
-    ac->setDefaultShortcut(m_ui->actionEntryDelete, Qt::CTRL + Qt::Key_D);
-    ac->setDefaultShortcut(m_ui->actionEntryDelete, Qt::Key_Delete);
```

Comment:
The second of these lines was overriding the first, so only Delete was actually bound to this action. Now that this shortcut is moved to the .ui file, it's still only Delete that's bound to the action, so no change in behaviour.

I don't know if there was an idea at some point to have Ctrl+D be a valid alternative delete shortcut.

Reply:
Ctrl+D should be used for Cloning (ie, duplicating)

Reply:
Cloning an entry is currently bound to Ctrl+K – see the line just below.

Reply:
This is just the default shortcut, you can add any number of shortcuts as alternates. Although I have never used Ctrl + K since it is entirely obtuse.

Reply:
You're saying you'd like to add Ctrl+D as an alternate shortcut for cloning? Perhaps that would be best done in a separate PR?

I'm also not sure how that would interact with the shortcut editor – that seems to only support a single shortcut per action.

Reply:
https://doc.qt.io/qt-5/qaction.html#setShortcuts

I'd be happy with changing the default for clone to Ctrl + D

Reply:
Yes, I had noticed that QAction supports multiple shortcuts. But the KeePassXC shortcut editor UI only supports assigning a single shortcut per action. So pre-defining multiple shortcuts for an action may not work well with that, is what I was getting at.

But ok, I can just change the default for cloning from Ctrl+K to Ctrl+D then (without using multiple shortcuts for that action).

Reply:
Yah I removed the ability to assign multiple shortcuts since it didn't work well.